### PR TITLE
only fetch MetafeedFinder after disk state loaded

### PR DESCRIPTION
**Pull request by staltz**
## Context

Stumbled upon this issue in https://github.com/ssbc/ssb-replication-scheduler/pull/11

## Problem

If you call MetafeedFinder `fetch` before it has loaded state (all the `metafeed/announce` msgs from the database), then it will ask remote peers for data that is already in our local database

## Solution

Queue all `fetch` calls until the database has loaded all the `'metafeed/announce'` messages.

### CODE REVIEW

File: metafeed-finder.js [hunk 1]
```diff
@@ -40,23 +59,25 @@ module.exports = class MetafeedFinder {
...
+            const metaFeedId = rootMF.id
+            this._map.set(mainFeedId, metaFeedId)
+            this._inverseMap.set(metaFeedId, mainFeedId)
+            cb(null, metaFeedId)
+          }
+        })
+      } else {
+        this._request(mainFeedId, cb)
+      }
+    })
```
Comment (staltz) [inline metafeed-finder.js:80]: Everything inside the `this._onLoaded.onReady(() => {` is just indented, but the diff makes it hard to see.

File: metafeed-finder.js [hunk 2]
```diff
@@ -9,6 +9,24 @@ const detectSsbNetworkErrorSeverity = require('ssb-network-errors')
 const { where, type, live, toPullStream } = require('ssb-db2/operators')
 const { validateMetafeedAnnounce } = require('ssb-meta-feeds/validate')
 
+class ReadyGate {
```
Comment (arj03) [inline metafeed-finder.js:12]: I smell a module coming up :)

**Review by arj03 — APPROVED** [review]
Nice find. I think I saw that readygate code somewhere else. I'm fine with it being here, just a general comment.

File: metafeed-finder.js (hunk 2 above)
Comment (staltz) [reply metafeed-finder.js:12]: Sigh, but not yet. I'm not too satisfied with this name, and with the whole implementation in general.

**Comment by staltz** [comment]
Yeah, lets eventually do something about that pattern. It's so little code right now, that it doesn't look like a technical debt.

**Pull request merged**